pkg/p2p/client_test.go

```diff
@@ -341,9 +341,11 @@ func TestClientSendAnomalies(t *testing.T) {
 		time.Sleep(100 * time.Millisecond)
 		closeClient()
 	}()
-	_, err = client.SendMessage(ctx, "test-topic", &testMessage{Value: 1})
-	require.Error(t, err)
-	require.Regexp(t, ".*ErrPeerMessageClientClosed.*", err.Error())
+	_, _ = client.SendMessage(ctx, "test-topic", &testMessage{Value: 1})
+	// There is no need to check for error here, because when a client is closing,
+	// message loss is expected because sending the message is fully asynchronous.
+	// The client implementation is considered correct if `SendMessage` does not
+	// block infinitely.
```

Comment:
Is the timeline like
timeline|-|-
-- | -- | --
t1 | client is closing |  
t2 |   | client.SendMessage, buffer SendChannelSize messages at most
t3 |   | client.SendMessage will block
t4 | client is closed |  
t5 |   | client.SendMessage returns ErrPeerMessageClientClosed

Reply:
Yes